pkg/build/api/validation/validation.go

```diff
@@ -445,6 +462,28 @@ func validateCustomStrategy(strategy *buildapi.CustomBuildStrategy, fldPath *fie
 	return allErrs
 }
 
+func validateJenkinsPipelineStrategy(strategy *buildapi.JenkinsPipelineBuildStrategy, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(strategy.JenkinsfilePath) != 0 && len(strategy.Jenkinsfile) != 0 {
+		return append(allErrs, field.Invalid(fldPath, strategy, "must provide a value for exactly one of jenkinsfilePath, or jenkinsfile"))
+	}
```

Comment:
I think we discussed before (here or on IRC) that empty would indicate the default of using Jenkinsfile in root of git repo. Happy to take your lead if you think it should be explicitly required.